android/KMEA/app/src/main/java/com/tavultesoft/kmea/data/CloudDownloadMgr.java

```diff
@@ -0,0 +1,238 @@
+   * @param aTargetModel the target model
+   * @param aCallback the callback
+   * @param params the cloud api params for download
+   * @param <ModelType> the target models type
+   * @param <ResultType> the cloud requests result type
+   */
+  public <ModelType,ResultType> void executeAsDownload(Context aContext, String aDownloadIdentifier,
+                                ModelType aTargetModel,
+                                ICloudDownloadCallback<ModelType,ResultType> aCallback,
+                                CloudApiTypes.CloudApiParam... params)
+  {
+    if(!isInitialized)
+      throw new IllegalStateException("Downloadmanager is not initialize. Call KMManger.initialize before");
+
+    synchronized (downloadSetByDownloadIdentifier) {
+
+      if(alreadyDownloadingData(aDownloadIdentifier))
+        return;
+
+      DownloadManager downloadManager = (DownloadManager) aContext.getSystemService(Context.DOWNLOAD_SERVICE);
+
+      CloudApiTypes.CloudDownloadSet<ModelType,ResultType> _downloadSet =
+        new CloudApiTypes.CloudDownloadSet<ModelType,ResultType>(
+        aDownloadIdentifier,aTargetModel);
+      _downloadSet.setCallback(aCallback);
+
+      downloadSetByDownloadIdentifier.put(aDownloadIdentifier,_downloadSet);
+
+      for(int _i=0;_i<params.length;_i++)
+      {
+        CloudApiTypes.SingleCloudDownload _download = createRequest(aContext,_i,params[_i]);
+        _download.setDownloadId(downloadManager.enqueue(_download.getRequest()));// enqueue puts the download request in the queue.
+        internalDownloadIdToDownloadIdentifier.put(_download.getDownloadId(),aDownloadIdentifier);
+        _downloadSet.addDownload(_download);
+      }
+    }
+
+  }
+
+  /**
+   * create request from api param.
+   * @param aContext the context
+   * @param aNo the number
+   * @param aParam the api parameter
+   * @return the single download
+   */
+  private CloudApiTypes.SingleCloudDownload createRequest(Context aContext, int aNo, CloudApiTypes.CloudApiParam aParam)
+  {
+
+    File _file=new File(aContext.getExternalFilesDir(null),"download_"+System.currentTimeMillis()+"_"+aNo);
+       /*
+       Create a DownloadManager.Request with all the information necessary to start the download
+        */
+    DownloadManager.Request _request=new DownloadManager.Request(Uri.parse(aParam.url))
+      .setTitle("Cloud Download " + aParam.target)
+      .setDescription("Downloading " + aParam.target)
+      .setNotificationVisibility(DownloadManager.Request.VISIBILITY_VISIBLE)// Visibility of the download Notification
+      .setDestinationUri(Uri.fromFile(_file));// Uri of the destination file
+      // api level 24
+      //.setRequiresCharging(false)// Set if charging is required to begin the download
+      //.setAllowedOverMetered(true)// Set if download is allowed on Mobile network
+      //.setAllowedOverRoaming(true);// Set if download is allowed on roaming network
```

Comment:
We probably want to check the SDK version number and apply these settings for API > 24.
e.g
```
if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {

}

Maybe setAllowedOverRoaming(false) though?
```

Reply:
I think this would be part of issue #2257
For now we keep the default properties the downloadmanager is providing.